portal-kernel/src/com/liferay/portal/kernel/json/JSONUtil.java

```diff
@@ -207,18 +207,16 @@ public static boolean isEmpty(JSONArray jsonArray) {
 		return false;
 	}
 
-	public static boolean isValid(String json) {
+	public static JSONObject isValid(String json) {
```

Comment:
@slnn I fully accept the idea that it is redundant. However, this change breaks the convention of our method names -- `isXXX` should return a boolean.

I think this is something like what we did in the past for `Map` usages. In the past, many people use `Map` like

```
if (map.containsKey(key)) {
    var value = map.get(key);
    // do things with the value
}
```
which makes the `Map` to calculate the hash of the key and lookup twice. We changed it to
```
var value = map.get(key);

if (value != null) {
    // do things with the value
}
```

I think you can go with this approach, and ask SF team to add an SF rule -- if people need to use the `JSONObject` created from a string, they should directly create it instead of checking valid with `isValid` first.

Reply:
cc @tinatian

Reply:
Yes, it does break the convention of our method names, so I rename it in next commit